src/zenml/integrations/whylogs/whylogs_utils.py

```diff
@@ -0,0 +1,120 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import datetime
+from typing import Dict, Optional
+
+import pandas as pd
+from whylogs import DatasetProfile  # type: ignore
+from whylogs.app import Session  # type: ignore
+
+from zenml.steps.step_context import StepContext
+
+
+class WhylogsContext(StepContext):
```

Comment:
The implementation as a separate context is really clean, however I do have one concern: What if the user requires multiple integrations (lets say whylogs and mlflow) in one step, would that require multiple contexts in the signature? (We currently only allow a single step context, but that is easily changed. I just fear that the step signature will get too messy)

Reply:
I share the same concern with @schustmi . Figuring out a way to be able to use multiple integrations with a single StepContext might be critical here.

Reply:
As a potential solution, we could append an object or a function to the generic `context` variable if certain integrations are activated, e.g.:

```python
context.whylogs  # monkeypatch
# or
context.get_integration('whylogs')
# or
context.get_whylogs()
```